caper/cromwell_metadata.py

```diff
@@ -0,0 +1,172 @@
+                    subworkflow = call['subWorkflowMetadata']
+                    sub_m = CromwellMetadata(subworkflow)
+                    sub_m.recurse_calls(
+                        fnc_call, parent_call_names=parent_call_names + (call_name,)
+                    )
+                else:
+                    fnc_call(call_name, call, parent_call_names)
+
+    def write_on_workflow_root(self, basename=DEFAULT_METADATA_BASENAME):
+        """Update metadata JSON file on metadata's output root directory.
+        If there is a subworkflow, nest it's metadata into main workflow's one
+
+        Args:
+            write_subworkflow:
+                Write metadata JSON file for subworkflows.
+        """
+        if 'workflowRoot' in self._metadata:
+            root = self._metadata['workflowRoot']
+            metadata_file = '/'.join([root, basename])
+
+            AutoURI(metadata_file).write(json.dumps(self._metadata, indent=4) + '\n')
+            logger.info('Wrote metadata file. {f}'.format(f=metadata_file))
+        else:
+            metadata_file = None
+            workflow_id = self._metadata['id'] if 'id' in self._metadata else None
+            logger.warning(
+                'Failed to write metadata file. workflowRoot not found. '
+                'wf_id={i}'.format(i=workflow_id)
+            )
+
+        return metadata_file
+
+    def troubleshoot(self, fileobj, show_completed_task=False, show_stdout=False):
+        """Troubleshoots a workflow.
+        Also, finds failure reasons and prints out STDERR and STDOUT.
+
+        Args:
+            fileobj:
+                File-like object to write troubleshooting messages to.
+            show_completed_task:
+                Show STDERR (or STDOUT) of completed tasks.
+            show_stdout:
+                Show failed task's STDOUT along with STDERR.
+        """
+        fileobj.write(
+            '* Started troubleshooting workflow: id={id}, status={status}\n'.format(
+                id=self.workflow_id, status=self.workflow_status
+            )
+        )
+
+        if self.workflow_status == 'Succeeded':
+            fileobj.write('* Workflow ran Successfully.\n')
+            return
+
+        if self.failures:
+            fileobj.write(
+                '* Found failures JSON object.\n{s}\n'.format(
+                    s=json.dumps(self.failures, indent=4)
+                )
+            )
+
+        def troubleshoot_call(call_name, call, parent_call_names):
+            status = call['executionStatus'] if 'executionStatus' in call else None
+            shard_index = call['shardIndex'] if 'shardIndex' in call else None
+            rc = call['returnCode'] if 'returnCode' in call else None
+            job_id = call['jobId'] if 'jobId' in call else None
+            stdout = call['stdout'] if 'stdout' in call else None
+            stderr = call['stderr'] if 'stderr' in call else None
+            run_start = None
+            run_end = None
```

Comment:
If this is dict could do something like:
```
status = call.get('executionStatus')
job_id = call.get('jobId')
```
etc.

Reply:
Fixed as you suggested.